WordPress/gutenberg
# ProgressBar: Use gray 300 for track color

## What?
This PR updates the `ProgressBar` indicator color to use the gray 300 instead of gray 100 for the progress bar track.

## Why?
Improves the color combinations when used in practice in #53032.

## How?
We're just adjusting the indicator color to use the gray 300 instead of gray 100 for the progress bar track.

## Testing Instructions
* Run `npm run storybook:dev`
* Navigate to http://localhost:50240/?path=/docs/components-experimental-progressbar--default
* Verify the progress bar works and looks well. 
* Play with the Theme controls (the pencil icon) and verify the combinations work well.
* Test it with #53032.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 

Before:
![Screenshot 2023-08-04 at 16 10 24](https://github.com/WordPress/gutenberg/assets/8436925/593b0cf1-ce96-4469-8c74-4a9c898b0581)

After:
![Screenshot 2023-08-04 at 16 09 00](https://github.com/WordPress/gutenberg/assets/8436925/f1a263a1-6760-4ebf-9e1f-5e7e63b1071a)
